File: oo-workshop/liskov-substitution/src/main/java/com/example/mobprogramming/problem/Move.java
package com.example.mobprogramming.problem;

public class Move {
    boolean batteryCharged;
    Car car = new Car();
    Bus bus = new Bus();
    Bicycle bicycle = new Bicycle();
    Skateboard skateboard = new Skateboard();

    public void startMoving(){
        car.startEngine(batteryCharged);
        bus.startEngine(batteryCharged);
        bicycle.startEngine(batteryCharged); // there's no engine to start!
        skateboard.startEngine(batteryCharged); // there's no engine to start!
    }
}

File: oo-workshop/tsexercises/src/liskovsubstitution/move.ts
import {Car} from './car'
import {Bus} from './bus'
import {Bicycle} from './bicycle'
import {Skateboard} from './skateboard'


export class Move {
    batteryCharged: boolean | undefined;
    car: Car = new Car();
    bus: Bus = new Bus();
    bicycle: Bicycle = new Bicycle();
    skateboard: Skateboard = new Skateboard();

    public startMoving(batteryCharged: boolean): void{
        this.car.startEngine(batteryCharged);
        this.bus.startEngine(batteryCharged);
        this.bicycle.startEngine(batteryCharged); // there's no engine to start!
        this.skateboard.startEngine(batteryCharged); // there's no engine to start!
    }
}
